### ISSUE TITLE
Locking problem with propfind and get

### ISSUE THREAD

**Issue by xyzzy42** [couling/WebDAV-Daemon#18]
I've setup nginx to proxy a virtual host to webdavd on the same host but another port.  This works fine with cadaver.
Creating files with macos as the client works, but hangs.  I don't know if it's related to the nginx proxy or not, but I think not.  As macos needs class 2 support for write access, and webdavd only supports class 1, obviously no one has tested it with macos without a proxy and found it to work.

It appears to work fine through the `PUT` command which creates the file ok.  Then there is a `PROPFIND` on the newly created file and webdavd doesn't respond.

Here's the logs from nginx:
```
1.2.3.4 - tpiepho [04/Jun/2021:16:34:35 -0700] "LOCK /davtest/testfile HTTP/1.1" 200 457 "-" "WebDAVFS/3.0.0 (03008000) Darwin/19.6.0 (x86_64)" "-"
1.2.3.4 - tpiepho [04/Jun/2021:16:34:35 -0700] "PUT /davtest/testfile HTTP/1.1" 201 0 "-" "WebDAVFS/3.0.0 (03008000) Darwin/19.6.0 (x86_64)" "-"
2021/06/04 16:35:35 [error] 1776945#0: *443 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 1.2.3.4, server: vhost.server.name, request: "PROPFIND /davtest/testfile HTTP/1.1", upstream: "http://127.0.0.1:30001/davtest/testfile", host: "vhost.server.name"
1.2.3.4 - tpiepho [04/Jun/2021:16:35:35 -0700] "PROPFIND /davtest/testfile HTTP/1.1" 504 167 "-" "WebDAVFS/3.0.0 (03008000) Darwin/19.6.0 (x86_64)" "-"
1.2.3.4 - tpiepho [04/Jun/2021:16:35:35 -0700] "UNLOCK /davtest/testfile HTTP/1.1" 204 0 "-" "WebDAVFS/3.0.0 (03008000) Darwin/19.6.0 (x86_64)" "-"
1.2.3.4 - tpiepho [04/Jun/2021:16:35:35 -0700] "PROPFIND /davtest/testfile HTTP/1.1" 207 460 "-" "WebDAVFS/3.0.0 (03008000) Darwin/19.6.0 (x86_64)" "-"
```

The key thing to see is that exactly 1 minute after the `PUT`, which worked with 201 and created the file, the `PROPFIND` fails with 504, gateway timeout, because webdavd failed to respond.  Then there is an `UNLOCK`, which works, and a `PROPFIND` of the same file, which works this time and macos, after the 1 minute timeout, is happy.

The log from webdavd:
```
Jun 04 2021 16:34:35 127.0.0.1 tpiepho 200 LOCK /davtest/testfile
Jun 04 2021 16:34:35 127.0.0.1 tpiepho 201 PUT /davtest/testfile
Jun 04 2021 16:35:35 127.0.0.1 tpiepho 500 UNLOCK /davtest/testfile
Jun 04 2021 16:35:35 127.0.0.1 tpiepho 207 PROPFIND /davtest/testfile
Jun 04 2021 16:35:35 127.0.0.1 tpiepho 207 PROPFIND /davtest/testfile
```

Notice here that webdavd responded to TWO `PROPFIND`s after the `UNLOCK`.  The ordering is wrong here, one should be before, one after.  Also notice 1st `PROPFIND` is one minute after the `PUT`.

I checked with wireshark while this was running, actual sequence was `LOCK PUT PROPFIND `(one minute delay)` UNLOCK PROPFIND`.

It seems like what is happening is the `PROPFIND` on a locked file is blocking until the file is unlocked.

**Comment by xyzzy42** [comment]
I've tracked this down to what I think is the cause.

webdavd is trying to acquire a lock when it processes PROPFIND and GET requests.  It shouldn't do that.

Clients that use locking, and I've tested macos as well as cadaver's edit commit, will lock a file and then use PROPFILE or GET on it without sending the lock uuid.  webdavd tries to get a shared lock in this case and hangs because there is already an exclusive lock on the file.

webdavd shouldn't worry about locks when processing PROPFIND or GET.  See [RFC4918 §7](http://www.webdav.org/specs/rfc4918.html#rfc.section.7.p.5):

> Of the methods defined in HTTP and WebDAV, PUT, POST, PROPPATCH, LOCK, UNLOCK, MOVE, COPY (for the destination resource), DELETE, and MKCOL are affected by write locks. All other HTTP/WebDAV methods defined so far -- GET in particular -- function independently of a write lock.

Once I removed this unnecessary locking it seems to work ok.
